CMakeLists.txt

```diff
@@ -5,18 +5,44 @@ cmake_minimum_required(VERSION 3.11)
 set(CCF_DIR ${CMAKE_CURRENT_SOURCE_DIR})
 include(${CCF_DIR}/cmake/preproject.cmake)
 
+# Deduce project version from git environment
+if(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/.git)
+  FIND_PACKAGE(Git)
+  execute_process(
+    COMMAND git describe --tags
+    OUTPUT_VARIABLE "CCF_VERSION"
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+  )
+  execute_process(
+    COMMAND bash "-c" "git describe --tags --abbrev=0 | cut -d'-' -f1 | tr -d v"
+    OUTPUT_VARIABLE "CCF_RELEASE_VERSION" OUTPUT_STRIP_TRAILING_WHITESPACE
+  )
+  # Check that git release tag follows semver
+  execute_process(
+    COMMAND
+      bash "-c"
+      "[[ ${CCF_RELEASE_VERSION} =~ ^([[:digit:]])+(\.([[:digit:]])+)*(-.*)?$ ]]"
```

Comment:
That prevents us from cutting -rc tags, that doesn't seem ideal?

Reply:
That's actually fine because `CCF_RELEASE_VERSION` does not include anything after the first `-`, e.g. if the tag is `1.0.0-rc1` then `CCF_RELEASE_VERSION` is `1.0.0`. `CCF_VERSION` on the other hand, used for `cchost --version` and the name of the CCF install directory would be `1.0.0-rc1`.